### PR TITLE
Deprecate `ProgressBar` and rename it to `TQDMProgressBar`

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Deprecate ProgressBar and rename it to TQDMProgressBar.

Follow up PR would be updating the tests and docs.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by rohitgr7 — APPROVED** [review]
should this be updated too?
https://github.com/PyTorchLightning/pytorch-lightning/blob/2d4d60934d1bb7ffcda5995ae7706c508e2ae64c/pytorch_lightning/trainer/trainer.py#L293-L308

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ def configure_progress_bar(self, refresh_rate=None, process_position=0):
         if len(progress_bars) == 1:
             progress_bar_callback = progress_bars[0]
         elif refresh_rate > 0:
-            progress_bar_callback = ProgressBar(refresh_rate=refresh_rate, process_position=process_position)
+            progress_bar_callback = TQDMProgressBar(refresh_rate=refresh_rate, process_position=process_position)
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/connectors/callback_connector.py:233]: here too: https://github.com/PyTorchLightning/pytorch-lightning/blob/2d4d60934d1bb7ffcda5995ae7706c508e2ae64c/pytorch_lightning/trainer/connectors/callback_connector.py#L82-L95
here too in the same file

**Comment by kaushikb11** [comment]
@awaelchli @Borda Would need any of your approval for this.

**Review by tchaton — APPROVED** [review]
LGMT !

**Pull request merged**